### ISSUE TITLE
feat: Create a runtime Capacitor addPlatform() api

### ISSUE THREAD

**Issue by adamdbradley** [ionic-team/capacitor#4243]
### Description

Right now ios and android are baked into the runtime, which is fine and will probably stay that way. However, it should be easier for other platforms to add themselves to the Capacitor runtime, such as Electron.

### Preferred Solution

Something along the lines of:

```
import { addPlatform } from '@capacitor/core';

addPlatform({
  name: 'electron',
  isNativePlatform: () => !!window.electron,
  ...
});
```

And this same API can be what's used for ios and android internally. I think we'll want to keep a map of all the platforms available, and when the top level API, such as `Capacitor.getPlatform()` is called, it loops through each of the platforms to see if it provided its own API or not, otherwise falling back to what it's doing today.

**Comment by IT-MikeS** [comment]
This API would also allow for the override of other platform functions such as `isPluginAvailable`. 

For example in electron, we would want electron Plugins first but all web ones also work just fine, so the above function should return electron first then web if possible.

**Comment by thomasvidas** [comment]
Merged in #4255
